runtime/j9vm/javanextvmi.cpp

```diff
@@ -274,21 +274,57 @@ enterVThreadTransitionCritical(J9VMThread *currentThread, jobject thread)
 	MM_ObjectAccessBarrierAPI objectAccessBarrier = MM_ObjectAccessBarrierAPI(currentThread);
 	j9object_t threadObj = J9_JNI_UNWRAP_REFERENCE(thread);
 
-	while(!objectAccessBarrier.inlineMixedObjectCompareAndSwapU64(currentThread, threadObj, vm->virtualThreadInspectorCountOffset, 0, (U_64)-1)) {
+retry:
+	while (!objectAccessBarrier.inlineMixedObjectCompareAndSwapU64(currentThread, threadObj, vm->virtualThreadInspectorCountOffset, 0, (U_64)-1)) {
 		/* Thread is being inspected or unmounted, wait. */
 		vmFuncs->internalReleaseVMAccess(currentThread);
 		VM_AtomicSupport::yieldCPU();
 		/* After wait, the thread may suspend here. */
 		vmFuncs->internalAcquireVMAccess(currentThread);
 		threadObj = J9_JNI_UNWRAP_REFERENCE(thread);
 	}
+
+	/* Link the current J9VMThread with the virtual thread object. */
+	if (!objectAccessBarrier.inlineMixedObjectCompareAndSwapU64(currentThread, threadObj, vm->isSuspendedInternalOffset, 0, (U_64)currentThread)) {
+		/* If virtual thread is suspended while unmounted, reset the inspectorCount and do a wait and retry. */
+		if (J9_ARE_ANY_BITS_SET(J9OBJECT_U64_LOAD(currentThread, threadObj, vm->isSuspendedInternalOffset), 1)) {
+			J9OBJECT_I64_STORE(currentThread, threadObj, vm->virtualThreadInspectorCountOffset, 0);
+		}
```

Comment:
Why do we need to reset the `inspectorCount`?

Reply:
if a vthread is suspended while unmounted, there will be no halt flag since the native vm thread is not linked.
when that thread is attempted to be mounted, we need to avoid acquiring either the inspector count or linking to vmthread so it is still viewed as a unmounted thread